sklearn/utils/validation.py

```diff
@@ -352,11 +352,6 @@ def check_array(array, accept_sparse=False, accept_large_sparse=True,
         to be any format. False means that a sparse matrix input will
         raise an error.
 
-        .. deprecated:: 0.19
-           Passing 'None' to parameter ``accept_sparse`` in methods is
-           deprecated in version 0.19 "and will be removed in 0.21. Use
-           ``accept_sparse=False`` instead.
```

Comment:
I wonder if we should keep this notice, the parameter is still there after all? The wording is not great, one cannot "remove passing 'None' to parameter ``accept_sparse``"

What happens when one passes `accept_sparse=None` with this change?

Reply:
If X is dense, accept_sparse is ignored as it was before.
If X is sparse, `accept_sparse=None` raises an informative error:
`Parameter 'accept_sparse' should be a string, boolean or list of strings. You provided 'accept_sparse=None'`
I think the behavior is good.